src/Functions/map.cpp

```diff
@@ -147,6 +149,99 @@ class FunctionMap : public IFunction
     }
 };
 
+// mapFromArrays(keys, values) is a function that allows you to make key-value pair from a pair of arrays
+class FunctionMapFromArrays : public IFunction
+{
+public:
+    static constexpr auto name = "mapFromArrays";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionMapFromArrays>();
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 2;
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 2)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Function {} requires 2 arguments, but {} given", getName(), arguments.size());
+
+        const auto * keys_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
+        if (!keys_type)
+            throw Exception{"First argument for function " + getName() + " must be a Array",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+        const auto * values_type = checkAndGetDataType<DataTypeArray>(arguments[1].get());
+        if (!values_type)
+            throw Exception{"Second argument for function " + getName() + " must be a Array",
```

Comment:
I am pretty sure that exceptions with the exception type (ILLEGAL_TYPE_OF_ARGUMENT) as argument were outlawed in the meantime. If you rebase from master, you would have to write it as

```cpp
throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument for function {} must be a Array", getName());
```